channel/handlers_client.go

```diff
@@ -1334,28 +1334,175 @@ func (server *Server) playerUseCash(conn mnet.Client, reader mpacket.Reader) {
 	slot := reader.ReadInt16()
 	itemID := reader.ReadInt32()
 
+	used := false
+
 	switch itemID {
 	case constant.ItemSafetyCharm:
+		// Safety charm prevents exp loss on death - stored in player state
+		// Implementation would require death handling modification
+		log.Printf("Safety Charm not fully implemented")
+		// Do not mark as used until implementation is complete
+
 	case constant.ItemAPReset:
+		// Read stat up and stat down from packet
+		statUp := reader.ReadInt32()
+		statDown := reader.ReadInt32()
+		
+		// Validate stats are valid
+		if (statUp == constant.StrID || statUp == constant.DexID || statUp == constant.IntID || statUp == constant.LukID) &&
+			(statDown == constant.StrID || statDown == constant.DexID || statDown == constant.IntID || statDown == constant.LukID) &&
+			statUp != statDown {
+			// Basic validation: ensure we have points to take from statDown
+			canReset := false
+			switch statDown {
+			case constant.StrID:
+				canReset = plr.str > 4
+			case constant.DexID:
+				canReset = plr.dex > 4
+			case constant.IntID:
+				canReset = plr.intt > 4
+			case constant.LukID:
+				canReset = plr.luk > 4
+			}
+
+			if canReset {
+				// Remove 1 from stat down
+				switch statDown {
+				case constant.StrID:
+					plr.setStr(plr.str - 1)
+				case constant.DexID:
+					plr.setDex(plr.dex - 1)
+				case constant.IntID:
+					plr.setInt(plr.intt - 1)
+				case constant.LukID:
+					plr.setLuk(plr.luk - 1)
+				}
+
+				// Add 1 to stat up
+				switch statUp {
+				case constant.StrID:
+					plr.setStr(plr.str + 1)
+				case constant.DexID:
+					plr.setDex(plr.dex + 1)
+				case constant.IntID:
+					plr.setInt(plr.intt + 1)
+				case constant.LukID:
+					plr.setLuk(plr.luk + 1)
+				}
+
+				used = true
+			}
+		}
+
 	case constant.ItemSPResetFirstJob, constant.ItemSPResetSecondJob, constant.ItemSPResetThirdJob:
+		// Read skill up and skill down from packet
+		skillUp := reader.ReadInt32()
+		skillDown := reader.ReadInt32()
+		
+		// Implementation would require skill validation and adjustment
```

Comment:
Can you implement this? Just iterate over the players skills struct and adjust as necessary, it's easy. This is normal logic for adding sp:
https://github.com/ErwinsExpertise/Valhalla/blob/master/channel/handlers_client.go#L586